### DISCUSSION TITLE
Validator and EditContext / EditForm

### DISCUSSION THREAD

**Discussion by AntonitoMt** [Megabit/Blazorise#2433] [General]
We are evaluating several control suites. We want to know if your validator component should be bound to an EditContext / EditForm (like the basic one in Blazor) or can it be used independently in any other form/inputs component.
Thank you for your help.

**Comment by stsrki** [comment]
Hi,

Blazorise `Validation` component is very flexible and can be used independently. You can place it anywhere, not just inside of a form like it is required in basic Blazor. With Blazorise validation you get three types of validations built-in:

- validator handler using methods
- data-annotations validator
- pattern validator

*In v0.9.4, a method validator will be able to have async handlers.

Also, you can define custom validation handlers and use those if built-in is not enough. Some people have implemented fluent validation like this.

I hope I answered you question :)

**Reply by AntonitoMt** [reply]
Great clarification and explanation.
Thank you and I wish you a lot of success with that excellent component library that is Blazorise.
